### DISCUSSION TITLE
How to create a DNS zone called _msdcs.mydomain.org

### DISCUSSION THREAD

**Discussion by linuxrrze** [sys4/netbox-plugin-dns#318] [General]
Hi,

I'm currently trying to migrate my current DNS config to netbox_dns. However as I'm running a samba domain controller there are a lot of underscore dns records required.

I managed to create most of them using the "tolerate_*" options from the documentation.

However I couldn't find a way to create a zone with a leading underscore (like _msdcs.mydomain.org).

Should this be possible and if so what config settings are required to do so?

Error message is "Not a valid DNS domain name" (below the "Name" input).

Any help is welcome.

Bye,
   Marcel

**Comment by peteeckel** [comment]
Hi @linuxrrze, 

Microsoft never fails to surprise, it just rarely does so in a pleasant way :-)

```
tolerate_characters_in_domain_labels = "_"
```

> Update: The correct variable name is `tolerate_characters_in_zone_labels`, thanks to @linuxrrze for pointing that out!

makes the underscore behave like an ordinary character. And it allows you to create your desired domain.

<img width="1051" alt="Screenshot 2024-07-16 at 00 11 55" src="https://github.com/user-attachments/assets/dc4ec9c0-e71f-432d-ac57-5427848fb692">

**Comment by linuxrrze** [comment]
Hi  @peteeckel,

thanks for your fast reply.

While researching some more I didn't find the option "tolerate_characters_in_domain_labels" in the documentation
   https://github.com/peteeckel/netbox-plugin-dns/blob/main/docs/using_netbox_dns.md
and there's also no (plain) match for it in the sources
   https://github.com/peteeckel/netbox-plugin-dns.git (main)

There's however the (documented) option "tolerate_characters_in_zone_labels"?

However by researching this I found the cause for my problems (and it's *very* basic):

I assumed that the docker container would update plugins automatically (which it does not).
So I was still running a 1.0.0 version of netbox-dns in there (with no notion of the options specified in the docs).

With version 1.0.4 and the following config underscore zones are no longer a problem:

```
   'netbox_dns': {
        'tolerate_underscores_in_labels': True,
        'tolerate_characters_in_zone_labels': "_",
        'tolerate_leading_underscore_types': ["TXT", "SRV", "CNAME", "NS"],
    },

```

Thanks again,
  Marcel

**Comment by peteeckel** [comment]
Hi @linuxrrze, apparently I shouldn't write answers in the middle of the night, `tolerate_characters_in_domain_labels ` was clearly my mistake. Sorry for that, but I'm glad it led you to the correct solution.

The `tolerate_characters_in_zone_labels` setting is quite new and was a result of #299. In this case it wasn't even the usual suspect for DNS atrocities but some RFC authors themselves who had the (IMHO terrible) idea of suggesting `/` as part of a zone label. Even though they advised against it immediately after, at least one provider obviously didn't read that far.

The `hostname` and `domain` terminology has been removed from the code as it is not really correct in DNS context.
